src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2298,7 +2307,7 @@ public bool PackageExistsInPackagesFolder(PackageIdentity packageIdentity)
         /// project <paramref name="nuGetProject" /> is also installed in any
         /// other projects in the solution.
         /// </summary>
-        public static async Task<bool> PackageExistsInAnotherNuGetProject(NuGetProject nuGetProject, PackageIdentity packageIdentity, ISolutionManager solutionManager, CancellationToken token)
+        public static async Task<bool> PackageExistsInAnotherNuGetProject(NuGetProject nuGetProject, PackageIdentity packageIdentity, ISolutionManager solutionManager, CancellationToken token, bool excludeIntegrated = false)
```

Comment:
OK, I've switched to an overload. Regarding that parameter - yes, both `xproj` and `csproj` with `project.json` will be ignored if that parameter is specified. I've renamed it `ignoreNuGetIntegratedProjects` (which relates to the fact that the project types in question implement `INuGetIntegratedProject`) - is that sufficient, or do you have any other suggestions?